### ISSUE TITLE
Updating a file pointed to in aws_emr_cluster.configurations should trigger instance recreation

### ISSUE THREAD

**Issue by synhershko** [terraform-providers/terraform-provider-aws#1385]
E.g.

```hcl
resource "aws_emr_cluster" "querying" {
  name          = "Querying"
  release_label = "emr-5.7.0"
  applications  = ["Hadoop", "Hive", "Presto", "Ganglia"]

  configurations = "querying-emr-configurations.json"

  master_instance_type = "m4.large"
  core_instance_type   = "m4.2xlarge"
  core_instance_count  = 2
}
```

Updating the file `querying-emr-configurations.json` isn't detected and doesn't trigger new instance creation while it should.

Using v0.10.0

**Comment by bflad** [comment]
The `configurations` attribute is quite an odd implementation in this older resource. Its taking in a file path/URL as an input while the API only ever would know how to return the JSON content. There are two issues here:

* During the resource read function, this attribute is never getting set into the Terraform state properly (the `d.Set()` call for it will always fail with trying to write the wrong `[]*emr.Configuration` type as `schema.TypeString`)
* Even if we fix the above, we cannot detect drift between the original input (a file path or URL) and what the API returns (JSON content)

It seems like the `configurations` attribute needs to be deprecated and a new attribute (`configurations_json`?) created that properly only accepts JSON input so it can be properly compared to what the API returns.

**Comment by bflad** [comment]
Related: #543, which would also be fixed with JSON <-> JSON attribute (as it wouldn't matter where the content was coming from to Terraform).

**Comment by bflad** [comment]
Version 1.30.0 of the AWS provider, releasing later today, will have a new `configurations_json` argument that accepts the actual JSON contents and can correctly determine drift from the EMR API. In an upcoming release we'll mark the existing `configurations` argument as deprecated.

To convert to the new attribute, given this Terraform configuration:

```hcl
resource "aws_emr_cluster" "example" {
  # ... other configuration ...
  configurations = "example.json"
}
```

Update it to:

```hcl
resource "aws_emr_cluster" "example" {
  # ... other configuration ...
  configurations_json = "${file("example.json")}"
}
```

**Comment by bflad** [comment]
This has been released in [version 1.30.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1300-august-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
